src/scval.js

```diff
@@ -191,20 +191,26 @@ export function nativeToScVal(val, opts = {}) {
       }
 
       return xdr.ScVal.scvMap(
-        Object.entries(val).map(([k, v]) => {
-          // the type can be specified with an entry for the key and the value,
-          // e.g. val = { 'hello': 1 } and opts.type = { hello: [ 'symbol',
-          // 'u128' ]} or you can use `null` for the default interpretation
-          const [keyType, valType] = (opts?.type ?? {})[k] ?? [null, null];
-          const keyOpts = keyType ? { type: keyType } : {};
-          const valOpts = valType ? { type: valType } : {};
-
-          return new xdr.ScMapEntry({
-            key: nativeToScVal(k, keyOpts),
-            val: nativeToScVal(v, valOpts)
-          });
-        })
+        Object.entries(val)
+          // The Soroban runtime expects maps to have their keys in sorted
+          // order, so let's do that here as part of the conversion to prevent
+          // confusing error messages on execution.
+          .sort(([key1], [key2]) => key1.localeCompare(key2))
```

Comment:
js maps allow null key, for the paranoid, is a null check on `key1` worthwhile?

Reply:
Good point! But that would be an invalid map for Soroban anyway, so... user error?

Reply:
sounds fine, it seemed`val` param is created by user, and not much convention over what gets passed in that case, if they send a map with a null key they'll get the message from the stack trace!

Reply:
Yep, exactly! This is all just conversion to smart contract values which users define anyway - if they pass null it isn't valid, so we can avoid the extra check!